### ISSUE TITLE
loading a match from PastMatches should write to textboxes

### ISSUE THREAD

**Issue by PatheticMustan** [PatheticMustan/ScoutingApp2019#36]
Right now loading a match from PastMatches doesn't load values for Textboxes.

This happens because of a performance fix.
Since textboxes update very quickly, it was causing some performance issues.

The other components work properly because they can write to Redux, and after an action, Redux can update the components.

The textboxes aren't updated because the redux value keeps getting overwritten by whatever's in the component state

**Comment by PatheticMustan** [comment]
Ah, I got it!

One solution would be to keep another state!

Before we update ReduxState with ComponentState, we first check if the ReduxState has been changed since we last checked. Let's call this Watchdog

There are only a few ways ReduxText can be updated.

1.  Using the textbox. (Watchdog is updated before the check)
2. PastMatches (Watchdog and ReduxText will NOT match.)

If Watchdog and ReduxText do NOT match, it's probably because of LoadMatch, so we can set ComponentState to ReduxState
